File: examples/ts/basic.ts
/**
 * Basic morphological analysis example
 *
 * Run:
 *   npx tsx examples/ts/basic.ts
 *
 * Requires: npm package built (yarn build)
 */
import { Suzume } from '../../dist/index.js';

const suzume = await Suzume.create();

// Analyze Japanese text
const morphemes = suzume.analyze('すもももももももものうち');

for (const m of morphemes) {
  console.log(`${m.surface}\t${m.pos}\t${m.baseForm}`);
}

// Detailed output
console.log('\nDetailed:');
for (const m of morphemes) {
  console.log({
    surface: m.surface,
    pos: m.pos,
    posJa: m.posJa,
    baseForm: m.baseForm,
    reading: m.reading,
    conjType: m.conjType,
    conjForm: m.conjForm,
  });
}

File: examples/cpp/basic.cpp
/**
 * Basic morphological analysis example
 *
 * Build (from project root):
 *   cmake -B build && cmake --build build --parallel
 *   g++ -std=c++17 -Isrc -Lbuild/lib -o basic examples/cpp/basic.cpp \
 *     -lsuzume -lsuzume_analysis -lsuzume_postprocess -lsuzume_grammar \
 *     -lsuzume_dictionary -lsuzume_core -lsuzume_normalize -lsuzume_pretokenizer
 */
#include <iostream>

#include "core/types.h"
#include "suzume.h"

int main() {
  suzume::Suzume analyzer;

  // Analyze Japanese text
  auto morphemes = analyzer.analyze("すもももももももものうち");

  for (const auto& m : morphemes) {
    std::cout << m.surface << "\t" << suzume::core::posToString(m.pos) << "\t"
              << m.getLemma() << "\n";
  }

  return 0;
}
